src/main/scala/sbtprotoc/ProtocPlugin.scala

```diff
@@ -144,9 +146,51 @@ object ProtocPlugin extends AutoPlugin {
 
   override def projectConfigurations: Seq[Configuration] = Seq(ProtobufConfig)
 
-  def protobufGlobalSettings: Seq[Def.Setting[_]] =
+  override def globalSettings: Seq[Def.Setting[_]] = protobufGlobalSettings
+
+  override def projectSettings: Seq[Def.Setting[_]] =
+    protobufProjectSettings ++ inConfig(Compile)(protobufConfigSettings) ++
+      inConfig(Test)(protobufConfigSettings)
+
+  private[this] val protobufGlobalSettings: Seq[Def.Setting[_]] =
     Seq(
+      PB.protocVersion := "3.11.4",
+      PB.deleteTargetDirectory := true,
+      PB.cacheClassLoaders := true,
+      PB.protocOptions := Nil,
+      PB.targets := Nil,
       includeFilter in PB.generate := "*.proto",
+      dependencyResolution in PB.generate := {
+        val log = streams.value.log
+
+        val rootDependencyResolution =
+          (dependencyResolution in LocalRootProject).?.value
+
+        rootDependencyResolution.getOrElse {
+          log.warn(
+            "Falling back on a default `dependencyResolution` to " +
+              "resolve sbt-protoc plugins because `dependencyResolution` " +
+              "is not set in the root project of this build."
+          )
+          log.warn(
+            "Consider explicitly setting " +
+              "`Global / PB.generate / dependencyResolution` " +
+              "instead of relying on the default."
+          )
+
+          import sbt.librarymanagement.ivy._
+          val ivyConfig = InlineIvyConfiguration()
+            .withResolvers(Vector(Resolver.defaultLocal, Resolver.mavenCentral))
+            .withLog(log)
+          IvyDependencyResolution(ivyConfig)
+        }
+      },
+      // prevent protoc binary cache stampede when initializing projects with many modules by serializing executions
+      concurrentRestrictions += Tags.limit(ProtocDownload, 1)
```

Comment:
for builds with more than one version, the binary won't be retrieved in parallel, but I think it's a small price to pay